File: Euler2.c
#include <stdio.h>

long int Fibonacci();

int main(){

	long int total = Fibonacci();
	printf("%lu\n", total);


	return 0;
}

long int Fibonacci(){

	long int num1 = 0, num2 = 1, fib, sum = 0;

	do{
		fib = num1 + num2;
		if(fib % 2 == 0)
			sum += fib;
		num1 = num2;
		num2 = fib;
	}while (fib < 4000000);

	return sum;
}

File: Euler2.f90
program Euler2
	integer*4:: total = 0 !declare the variables as 4-byte integers
	integer*4:: num1 = 0
	integer*4:: num2 = 1
	integer*4:: fib = 0

	do while (fib < 4000000)
		fib = num1 + num2
		if (modulo(fib, 2) == 0) then !check if current fib is even
			total = total + fib
		end if
		num1 = num2
		num2 = fib
	end do
	print *, total
end program Euler2
